### PR TITLE
jenkins-cluster-arm

### PR DESCRIPTION
**Pull request by mahendra-shinde**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,596 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
```
Comment (bmoore-msft) [inline 101-jenkins-cluster-2-linux-1-win/azuredeploy.json:5]: format the json - e.g. SHIFT+ALT+F in VS Code

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+            "type": "string",
+            "defaultValue": "Standard_D2s_v3",
+            
+            "metadata": {
+                "description": "Windows Worker VM Size"
+            }
+        },
+        "master-username": {
+            "type": "string",
+            "defaultValue": "mahendra",
```
Comment (bmoore-msft) [inline 101-jenkins-cluster-2-linux-1-win/azuredeploy.json:39]: remove defaultValue

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+            "metadata": {
+                "description": "Admin Username for Jenkins Master"
+            }
+        },
+        "node-username": {
+            "type": "string",
+            "defaultValue": "mahendra",
```
Comment (bmoore-msft) [inline 101-jenkins-cluster-2-linux-1-win/azuredeploy.json:46]: remove defaultValue

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,596 @@
...
+        },
+        "node-password": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Password for slave instances"
+            }
+        },
+        "jenkins-dns": {
+            "type": "string",
+            "defaultValue": "jenkins-cluster-mx",
```
Comment (bmoore-msft) [inline 101-jenkins-cluster-2-linux-1-win/azuredeploy.json:65]: this must be globally unique, e.g.

[concat('jenkins-', uniqueString(resourceGroup().id))]

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,596 @@
...
+            "metadata": {
+                "description": "DNS Prefix for Jenkins Master"
+            }
+        },
+        "storageName":{
+            "type": "string",
+            "defaultValue": "jenkins123456",
```
Comment (bmoore-msft) [inline 101-jenkins-cluster-2-linux-1-win/azuredeploy.json:72]: this must be globally unique

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,596 @@
...
+            "metadata":{
+                "description": "storage account prefix"
+            },
+            "minLength":3,
+            "maxLength":20
+        }
+    },
+    
+    "functions": [],
```
Comment (bmoore-msft) [inline 101-jenkins-cluster-2-linux-1-win/azuredeploy.json:81]: remove this property

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,596 @@
...
+    "variables": {
+        "NetIpRange": "10.0.0.0/16",
+        "SubnetRange": "10.0.1.0/24",
+        "scriptUrlPrefix": "https://raw.githubusercontent.com/mahendra-shinde/azure-quickstart-templates/master/101-jenkins-cluster-2-linux-1-win"
```
Comment (bmoore-msft) [inline 101-jenkins-cluster-2-linux-1-win/azuredeploy.json:85]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,596 @@
...
+            "type": "Microsoft.Network/publicIPAddresses",
+            "apiVersion": "2019-11-01",
+            "location": "[parameters('location')]",
+            "tags": {
+                "displayName": "PublicIPAddress"
+            },
+            "properties": {
+                "publicIPAllocationMethod": "Dynamic",
+                "dnsSettings": {
+                    "domainNameLabel": "[toLower('node-1-jenkins-mx')]"
```
Comment (bmoore-msft) [inline 101-jenkins-cluster-2-linux-1-win/azuredeploy.json:274]: this must be globally unique

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,596 @@
...
+            "type": "Microsoft.Network/publicIPAddresses",
+            "apiVersion": "2019-11-01",
+            "location": "[parameters('location')]",
+            "tags": {
+                "displayName": "PublicIPAddress"
+            },
+            "properties": {
+                "publicIPAllocationMethod": "Dynamic",
+                "dnsSettings": {
+                    "domainNameLabel": "[toLower('node-2')]"
```
Comment (bmoore-msft) [inline 101-jenkins-cluster-2-linux-1-win/azuredeploy.json:395]: this must be globally unique

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.parameters.json [hunk 10]
```diff
@@ -0,0 +1,33 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
```
Comment (bmoore-msft) [inline 101-jenkins-cluster-2-linux-1-win/azuredeploy.parameters.json:4]: remove the parameters for which you have defaultValues in the template... use proper GEN values for the rest, see: https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#azure-devops-ci

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.json (hunk 1 above)
Comment (mahendra-shinde) [reply 101-jenkins-cluster-2-linux-1-win/azuredeploy.json:5]: fixed as said

File: 101-jenkins-cluster-2-linux-1-win/azuredeploy.parameters.json (hunk 10 above)
Comment (mahendra-shinde) [reply 101-jenkins-cluster-2-linux-1-win/azuredeploy.parameters.json:4]: fixed

**Pull request merged**